Tests/LiquidProjections.NHibernate.Specs/NHibernateProjectorSpecs.cs

```diff
@@ -1197,6 +1195,43 @@ private class ChildProjectionState
                 public bool Entry2Exists { get; set; }
             }
         }
+
+        public class When_event_handling_fails :
+            Given_a_sqlite_projector_with_an_in_memory_event_source
+        {
+            private const string originalExceptionMessage = "Some fail.";
+
+            public When_event_handling_fails()
+            {
+                Given(() =>
+                {
+                    Events.Map<CategoryDiscontinuedEvent>().As((@event, context) =>
+                    {
+                        throw new InvalidOperationException(originalExceptionMessage);
+                    });
+
+                    StartProjecting();
+                });
+            }
+
+            [Fact]
+            public async Task Then_it_should_throw_projection_exception()
+            {
+                ProjectionException exception = await Assert.ThrowsAsync<ProjectionException>(
+                    () => The<MemoryEventSource>().Write(UseThe(new CategoryDiscontinuedEvent())));
+
+                exception.Projector.Should().Be(typeof(ProductCatalogEntry).ToString());

```

Comment:
I would probably split the current event, the transaction batch and the innerexception into three *facts*. You can also make the tests more readable by comparing the properties with `ShouldBeEquivalentTo`. Same for the RavenDB specs.